pkg/decoder/delimiter.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package decoder
+
+import (
+	"regexp"
+	"sync"
+	"time"
+)
+
+// Line represents content separated by two '\n'
+type Line struct {
+	content []byte
+}
+
+// NewLine returns a new Line
+func NewLine(content []byte) *Line {
+	return &Line{
+		content: content,
+	}
+}
+
+// Delimiter
+type Delimiter interface {
+	LineChan() chan *Line
+}
+
+// SingleLineDelimiter reads lines from lineChan and uses lineBuffer to send them
+type SingleLineDelimiter struct {
+	lineChan   chan *Line
+	lineBuffer *LineBuffer
+}
+
+// NewSingleLineDelimiter returns a new NewSingleLineDelimiter
+func NewSingleLineDelimiter(outputChan chan *Output) *SingleLineDelimiter {
+	lineChan := make(chan *Line)
+	lineBuffer := NewLineBuffer(outputChan)
```

Comment:
I'm not sure why we need a buffer for `SingleLineDelimiter`

Reply:
And I'd avoid it even more as we copy again the data in the buffer

Reply:
`LineBuffer` keeps "...TRUNCATED..." in memory and offers a set of methods used by both delimiters but you right, there is a memory print drawback